Kernel/scheduler.c

```diff
@@ -116,7 +116,8 @@ int sch_unblockProcess(TPid pid) {
     TProcessState* processState;
     if (!tryGetProcessState(pid, &processState))
         return 1;
-
+    
+    // this check is unnecessary
     if (processState->status == READY)
         return 0;
```

Comment:
(I put this check there very early expecting that if the process wasn't ready, some "add to ready list" operation would have to occur. I'd just leave it as it is until we decide if we're gonna leave the scheduler as it is right now or implement a proper RR)